.github/workflows/publish.yml

```diff
@@ -2,59 +2,69 @@ name: publish
 
 on:
   push:
-    branches:
-      - main
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
   workflow_dispatch:
 
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: github-pages
+  cancel-in-progress: true
+
 jobs:
   build:
     runs-on: ubuntu-latest
+    # Run all steps inside the Hugo + Node container
+    container:
+      image: floryn90/hugo:ext-alpine
+      options: --user 0
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: true # Fetch Hugo themes (true OR recursive)
-          fetch-depth: 0 # Fetch all history for .GitInfo and .Lastmod
-      - uses: actions/setup-node@v4
-      - run: npm i -D postcss postcss-cli autoprefixer
-      - name: Setup Hugo
-        uses: peaceiris/actions-hugo@v3
-        with:
-          hugo-version: "0.134.1"
-          extended: true
+      - name: Checkout
+        uses: actions/checkout@v4
 
-      - name: Build
-        run: hugo --minify
+      # Computes the correct base URL
+      - name: Configure Pages
+        id: pages
+        uses: actions/configure-pages@v5
 
-      - name: Archive artifact
-        shell: sh
+      # Git is needed for Hugo Modules resolution
```

Comment:
I checked and looks like git is already installed in the `hugo:ext-alpine` image ([it's also used in its dockerfile](https://github.com/floryn90/docker-hugo/blob/7064a7ae51ae6165d58d91dfe0f4ff4b9742afc9/src/docker/alpine-ext/ext-alpine.df))